pkg/routeagent_driver/handlers/kubeproxy/endpoint_handler.go

```diff
@@ -71,7 +72,48 @@ func (kp *SyncHandler) LocalEndpointCreated(endpoint *submV1.Endpoint) error {
 	return nil
 }
 
-func (kp *SyncHandler) LocalEndpointUpdated(_ *submV1.Endpoint) error {
+func (kp *SyncHandler) LocalEndpointUpdated(endpoint *submV1.Endpoint) error {
+	if kp.localEndpoint == nil || kp.localEndpoint.Name != endpoint.Name {
```

Comment:
To be consistent, we should protect access to `localEndpoint` with the `syncHandlerMutex`.